pypdf/_writer.py

```diff
@@ -235,9 +235,9 @@ def _get_clone_from(
                 or Path(str(fileobj)).stat().st_size == 0
             ):
                 cloning = False
-            if isinstance(fileobj, (IO, BytesIO)):
+            if isinstance(fileobj, (IOBase, BytesIO)):
                 t = fileobj.tell()
-                fileobj.seek(-1, 2)
+                fileobj.seek(0, 2)
```

Comment:
Why do we need this change?

Reply:
Two reasons.  First, the original code was not correctly detecting an empty file; it was detecting a 1-byte file.  Second, to avoid an exception when `fileobj` refers to an empty file.  I can also tighten up the next line of code, as `seek()` returns the resulting file offset.

Reply:
Thanks for the explanation. Is this something we can actually test?

Reply:
I believe the second stanza of `test_stream_not_closed` tests this case.  I remember I had to make this fix in order to get tests working.  As far as the behavior of `seek()` I don't intend to test that directly.  Cpython calls `lseek()` which is well-documented and stable.  The OS cannot seek to absolute offset -1 of a file, which is what `lseek(fd, -1, SEEK_END)` is asking if `fd` refers to a zero-byte file.